internal/vault/vault.go

```diff
@@ -196,7 +205,7 @@ func (s *Store) Create(key, value string) error {
 
 	// We try to check whether key exists on the K/V store.
 	// If so, we must not overwrite it.
-	location := fmt.Sprintf("/kv/%s/%s", s.Location, key)
+	location := path.Join(s.Engine, s.Location, key) // /<engine>/<location>/<key>
```

Comment:
Isn't `/` in beginning needed this is not present with path.Join()

Reply:
The Vault client takes care of this. It will create a request similar to:
```
https://127.0.0.1:8200/secrets/prefix/my-key
```
So,if there is a missing `/` between the endpoint and the path it will insert one (part of request construction).

Reply:
Otherwise it would also not work if you e.g. specify `https://127.0.0.1:8200/` instead of `https://127.0.0.1:8200`. (Trailing `/`). Either you would have a missing `/` or 2 `/`s.